pytest-selenium/pages/content.py

```diff
@@ -0,0 +1,45 @@
+from selenium.webdriver.common.by import By
+from selenium.webdriver.support import expected_conditions as expected
+
+from pages.base import Page
+from regions.base import Region
+
+
+class Content(Page):
+    URL_TEMPLATE = "/books/{book_slug}/pages/{page_slug}"
+
+    _main_content_locator = (
+        By.CSS_SELECTOR,
+        '#main-content > div > div > div h1[data-type="document-title"]',
+    )
+    _table_of_contents_button_locator = (By.CSS_SELECTOR, "#root > div > div > button")
+    _table_of_contents_nav_locator = (By.CSS_SELECTOR, "#root > div > div > div > div > nav")
+
+    @property
+    def loaded(self):
+        return self.find_element(*self._main_content_locator)
```

Comment:
Does `self.find_element` throw an error when the locator matches `0` or `> 1` elements? If not, would it make sense to add a help that does that?

Reply:
Even when the locator matches `> 1` elements? I'm hoping that selenium can:

- help ensure there is exactly 1 `h1` on the page
- explicitly fail when more than 1 element matches a selector

Reply:
We would need to write a test for that. Selenium doesn't enforce any standards like that.

Reply:
What could be automated is some sort of HTML validation. I don't know any tools that do that off the top of my head but i'm sure something could be integrating. It would be something very similar to an aXe report. We would have it fail when there were any failures or some threshold.

Reply:
Something like this seems like it would do what you want. https://validator.w3.org/docs/api.html

Reply:
(just bike-shedding)

Oh, I guess that I was trying to say that if the test-writer expects the selector to match one element (by using `find_element` rather than `find_elements`) then it seems like that assumption should throw an error if more than one element is matched. Otherwise, `find_element` really means `find_first_element`.

Once I contribute a test that uses `find_element` I might refactor all of the `find_element` references to be something like `util.find_element(self, ...)` or something.